app/components/SignupForm.tsx

```diff
@@ -1,32 +1,111 @@
 "use client";
-import React from "react";
+import React, { useRef, useState } from "react";
 
 const SignupForm: React.FC = () => {
+  const initialValue = {
+    id: "",
+    name: "",
+    email: "",
+    pwd: "",
+    pwdConfirm: "",
+  };
+  const [errorMessage, setErrorMessage] = useState<{
+    [key: string]: string | undefined;
+  }>(initialValue);
+  const idRef = useRef<HTMLInputElement>(null);
+  const nameRef = useRef<HTMLInputElement>(null);
+  const emailRef = useRef<HTMLInputElement>(null);
+  const pwdRef = useRef<HTMLInputElement>(null);
+  const pwdConfirmRef = useRef<HTMLInputElement>(null);
+
+  const transInputValueToErrorMsg = (
+    key: string,
+    value: string,
+    validate?: {
+      required?: boolean;
+      min?: number;
+      max?: number;
+    }
+  ) => {
+    let errorMsg = "";
+    let emailRegex = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/i;
+    let pwdRegex = /^[A-Za-z0-9]*$/;
+
+    if (validate?.required && value === "") {
+      return (errorMsg = "값을 입력해주세요");
```

Comment:
```suggestion
      return "값을 입력해주세요."
```
* 여기서 굳이 errorMsg를 사용할 이유가 없어보입니다. 단순히 문자열을 리턴해도 됩니다. 코드가 단순해집니다.
* 마침표가 빠져있습니다.